File: sdk/generator/python/template/polar/literals.py
import typing

{% for enum in ir.enums %}
{{ enum.name }}: typing.TypeAlias = typing.Literal[
{%- for value in enum.values %}
{{ "%r" | format(value.value) }}{% if not loop.last %}, {% endif %}
{% endfor -%}
]
{% if enum.description %}
"""{{ enum.description }}"""
{% endif %}
{% endfor %}

File: sdk/generator/typescript/template/src/version/models/literals.ts
{% for enum in enums %}
/**
 * {{ enum.description or enum.name }}
 */
export type {{ enum.name }} = {% for value in enum.values %}{% if value.value is string %}"{{ value.value }}"{% elif value.value is boolean %}{{ value.value | lower }}{% else %}{{ value.value }}{% endif %}{% if not loop.last %} | {% endif %}{% endfor %};
{% endfor %}
